apps/src/code-studio/components/ModelManagerDialog.jsx

```diff
@@ -67,21 +110,49 @@ export default class ModelManagerDialog extends React.Component {
   };
 
   handleChange = e => {
-    this.setState({selectedModelId: e.target.value});
+    const model = this.getModelById(e.target.value);
+    this.setState({selectedModel: model});
+  };
+
+  showDeleteConfirmation = () => {
+    this.setState({confirmDialogOpen: true});
+  };
+
+  closeConfirmDialog = () => {
+    this.setState({confirmDialogOpen: false, deletionStatus: undefined});
+  };
+
+  deleteModel = () => {
+    $.ajax({
+      url: `/api/v1/ml_models/${this.state.selectedModel.id}`,
+      method: 'DELETE'
+    }).then(response => {
+      if (response.status === 'failure') {
+        this.setState({
+          deletionStatus: `Model with id ${response.id} could not be deleted.`
+        });
+      } else {
+        this.setState({confirmDialogOpen: false});
+      }
+    });
   };
 
   render() {
     const {isOpen} = this.props;
-    const noModels = this.state.models.length === 0;
+    const noModels =
+      !this.state.isModelListPending && this.state.models.length === 0;
```

Comment:
@breville this fixes the weirdness we saw where the warning popped up before the models were loaded